File: Models/DomainModels/CustomerDetail.cs
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;

namespace LenzPerson.api.Models.DomainModels
{
    public class CustomerDetail
    {
        public int Id { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        public string Password { get; set; }

        [Required]
        [RegularExpression(@"^[0-9]+$", ErrorMessage = "Phone number should contain only numbers.")]
        [StringLength(15, MinimumLength = 10, ErrorMessage = "Phone number should be between 10 and 15 digits.")]
        public string PhoneNumber { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        public string Gender { get; set; }

        [Required]
        public string City { get; set; }



        [Required]
        public string UserRole { get; set; }

        
    }
}

File: src/app/models/CustomerDetail.ts
export class CustomerDetail {
    id: number;
    email: string;
    password: string;
    phoneNumber: string;
    firstName: string;
    lastName: string;
    gender: string;
    city: string;
    userRole: string
   
  
    constructor(
      id: number,
      email: string,
      password: string,
      phoneNumber: string,
      firstName: string,
      lastName: string,
      gender: string,
      city: string,
      userRole: string
     
    ) {
      this.id = id;
      this.email = email;
      this.password = password;
      this.phoneNumber = phoneNumber;
      this.firstName = firstName;
      this.lastName = lastName;
      this.gender = gender;
      this.city = city;
      this.userRole =userRole;

    
    }
  }
  